### ISSUE TITLE
Regional steps output variables availability

### ISSUE THREAD

**Issue by cghanapu** [Optum/runiac#43]
We have a scenario where we need to access regional step's output named 'foo' in the following step which runs only in primary region. It would be handy if we can get hold of collection of 'foo' output variable generated from each regional step or output prefixed with region name and passed on to following step.

Ex:- <stepname>_foo = ['bar','bar2']
Ex:- <stepname>_<region_name>_foo = "bar"

**Comment by tiny-dancer** [comment]
This wouldn't be possible in current flow as all primary step deployments execute prior to the first regional step deployment.

What is the use case?

**Comment by cghanapu** [comment]
Agreed! We want to use the output variables generated in regional step in the next step but not in the same step.  
For instance, step1(foo) regional output variables availability in step2 (bar). 

├── step1_foo
├──--- regional
├──--------*.tf
├── step2_bar
├──------*.tf

**Comment by tiny-dancer** [comment]
Do you mind sharing the use case in regards to what you are trying to accomplish from a cloud perspective?
